### PR TITLE
first pass at travis-ci support

### PR DESCRIPTION
**Pull request by halkeye**
you can sorta see it working at https://travis-ci.org/halkeye/lasso

if you install the cli, its pretty easy to setup:

1) cd <path/to/vouch-proxy>
2) travis enable
3) travis set DOCKER_USERNAME <foo>
4) travis set DOCKER_PASSWORD <bar>

You can also do it from the GUI, which https://docs.travis-ci.com/user/environment-variables/#defining-variables-in-repository-settings but I can't check from this computer to give you exact steps

### CODE REVIEW

**Comment by bnfinet** [comment]
fantastic stuff @halkeye 

Right now the voucher/vouch-proxy repo is setup as an automated build.  I'm uncertain if `.travis/docker_push` is a better approach for publishing the Docker images.  What would you advise?

**Comment by halkeye** [comment]
So the only reason to do CI (travis/jenkins/codeship/whatevs) do the docker image, is if you wanted previous steps to pass before publishing.

Like go test has to be successful, then you publish the builds to docker hub

I can add `./do.sh test` to the docker build since its only in the builder image, not the published one.

File: do.sh [hunk 1]
```diff
@@ -83,30 +83,22 @@ goget () {
...
+  export EXTRA_TEST_ARGS='-cover'
+  test
   go tool cover -html=coverage.out -o coverage.html
 }
 
 test () {
-  if [ -n LASSO_CONFIG ]; then
-    export LASSO_CONFIG=${GOPATH}/src/github.com/LassoProject/lasso/config/test_config.yml
+  if [ -z "$VOUCH_CONFIG" ]; then
+    export VOUCH_CONFIG="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/config/test_config.yml"
```
Comment (bnfinet) [inline do.sh]: the `cd dirname ${BASH_SOURCE}` is already accomplished at the top of ./do.sh
https://github.com/vouch/vouch-proxy/blob/master/do.sh#L4-L8

Comment (halkeye) [reply do.sh]: Fixed.

**Comment by halkeye** [comment]
Merged master in as well

**Pull request merged**